lib/spack/spack/test/config.py

```diff
@@ -625,6 +625,53 @@ def test_add_command_line_scopes(tmpdir, mutable_config):
     spack.config._add_command_line_scopes(mutable_config, [str(tmpdir)])
 
 
+@pytest.mark.nomockstage
```

Comment:
Since this test doesn't override the build stage, I don't think `nomockstage` is required here.

Likewise with `test_alternate_override`.

Reply:
IMO there are two reasons for using `nomockstage` at this point:

1. avoid test failures
2. avoid the (accumulated) overhead of creating unnecessary mock stage directories on the filesystem

Reply:
Removed in latest commit.